codegen/jennies/openapi.go

```diff
@@ -0,0 +1,164 @@
+package jennies
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"path/filepath"
+	"strings"
+
+	"github.com/grafana/codejen"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/gengo/v2"
+	"k8s.io/gengo/v2/generator"
+	"k8s.io/kube-openapi/cmd/openapi-gen/args"
+	"k8s.io/kube-openapi/pkg/generators"
+
+	"github.com/grafana/grafana-app-sdk/codegen"
+)
+
+// OpenAPI jenny uses kube-openapi to build OpenAPI spec.
+// Because k8s.io/gengo doesn't allow you to use in-memory files (as the Overlay in the config is not accessible),
+// this jenny must be run _after_ the go type has been written to disk.
+type OpenAPI struct {
```

Comment:
NIT – how much effort would it be for use to just generate k8s-specific OAPI registration code ourselves, without resorting to using `k8s.io/gengo`?

I'm asking because we already have OAPI on our hands during CUE -> Go codegen (since we technically do CUE -> OAPI -> Go) and it would make sense to generate k8s OAPI files as part of normal Go codegen.

That'd let us get rid of the awkward post-processing step.

That said, I think it's totally fine to use post-processing approach for now and keep it that way if it's too much work to get off of it.

Reply:
I don't think it would be a impassible hurdle, but it's definitely more work (IIRC the naming of the OpenAPI needs to match the go packaging, and there were a few other gotchas that made it simper to just use `gengo` for the first pass). Long-term we can definitely explore migrating to doing all the codegen ourselves (while possibly still including the k8s gengo comments for if folks want to use them).

Reply:
Alright, sounds good. I _think_ we may want to add an issue to follow up on this but I'll leave it up to you to decide 👍